### PR TITLE
bpo-30607: Use external python-doc-theme

### PR DESCRIPTION
**Pull request by theacodes**
I have uploaded a build of these docs at http://temp.theadora.io/cpython/index.html

/cc @ncoghlan @brettcannon 

https://bugs.python.org/issue30607

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://cpython-devguide.readthedocs.io/pullrequest.html#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

**Comment by theacodes** [comment]
(I just signed the CLA, I'll nag the bot tomorrow when hopefully it'll have been reviewed.)

**Comment by brettcannon** [comment]
Just an FYI @jonparrott I just added the CLA bot to the theme repo.

File: .travis.yml
Comment (brettcannon) [inline .travis.yml]: Will there need to be a change made to the docs.python.org build step for the docs to pull the theme in?

Comment (ned-deily) [reply .travis.yml]: Also need to review how this change might affect installer builds and release builds which build copies of the docs.  Assigned myself to review.

Comment (theacodes) [reply .travis.yml]: @brettcannon probably, but I don't know where that build step is.

Comment (zware) [reply .travis.yml]: @jonparrott See https://github.com/python/docsbuild-scripts

Comment (theacodes) [reply .travis.yml]: @zware yes, we'll just need to add the theme package to the requirements.txt there. I'll send a PR.

Comment (theacodes) [reply .travis.yml]: Actually before I do that- @ncoghlan you mentioned that you didn't think we should publish the theme to PyPI, but it's seeming like it's going to be a lot more convenient to do that if we're gonna use it here. WDYT?

Comment (ncoghlan) [reply .travis.yml]: @jonparrott The main problem I see with it is that PyPI doesn't have great "teams" support, so it adds an awkward extra set of permissions to manage in an already awkward update process.

Whereas if we're installing the theme directly from git, then the CPython docs maintainers should already have all the access they need to update it - they'll just be modifying a different repo, rather than having to worry about also tagging that repo and making a new PyPI release.

Comment (theacodes) [reply .travis.yml]: @ncoghlan great point. I like it, I just wanted to understand why. I can add this to the comments here and in the build scripts. :)

**Comment by theacodes** [comment]
CLA signed. :)

I've also created https://github.com/python/docsbuild-scripts/pull/12

**Comment by jonparrott** [comment]
I'm happy to rebase this and keep it current, but this kind of died of lack of attention.

**Comment by zware** [comment]
With python/docsbuild-scripts#12 merged, I don't think there's anything blocking this now.

**Comment by jonparrott** [comment]
Thanks @zware, I'll attempt to rebase this (& reconcile any theme changes) over the weekend.

**Review by ned-deily — CHANGES_REQUESTED** [review]
See my comments on https://github.com/python/python-docs-theme/issues/5 about unnecessarily adding a docs build dependency on Git.  Also, the PR needs to be rebased and retested to accommodate the changes made to support multiple languages.  At that time, the full docs build should be tested ("make dist").   The step needed to install the theme package (presumably now from PyPI rather than from Github) should be added to the Makefile's "venv" recipe ("make venv") which currently installs the complete Docs build toolchain (now blurb, sphinx, and dependencies).

**Comment by jonparrott** [comment]
Alright, this is rebased and ready to review & merge. Thanks @ned-deily and @zware.

**Comment by ncoghlan** [comment]
Note that https://pypi.python.org/pypi/python-docs-theme/ is now 0.0.1, and there's only one version, so we can postpone questions of whether or not we want to pin versions on different CPython branches or not.

**Comment by jonparrott** [comment]
Gentle ping on this - I'd like to get it merged & hopefully do so before the branch gets stale. :)

**Review by ned-deily — APPROVED** [review]

**Comment by ned-deily** [comment]
Thanks for addressing my review comment; distributing via PyPI looks much cleaner to me.

@JulienPalard if you have the time and interest, I would appreciate your review of this, especially if there any implications for the translated docs.  Otherwise, we can deal with fixes during the 3.8 feature dev cycle (this change is not appropriate for 3.7 or earlier systems).  One other thing - the daily docs build system may need a change to install python-docs-theme.

**Comment by jonparrott** [comment]
> One other thing - the daily docs build system may need a change to install python-docs-theme.

@ned-deily is that what I did in https://github.com/python/docsbuild-scripts/pull/12

**Comment by ned-deily** [comment]
> is that what I did in python/docsbuild-scripts#12?

Ah, yes.  That should be updated to require the new package rather than getting from Github.

**Comment by jonparrott** [comment]
@ned-deily PR created: https://github.com/python/docsbuild-scripts/pull/41 (I don't think that blocks this)

**Comment by JulienPalard** [comment]
Merged #41 of docsbuild-scripts after testing it locally.

**Comment by jonparrott** [comment]
Thanks @JulienPalard. Anything else we need to do to get this merged? :)

**Comment by Mariatta** [comment]
Tested locally, things looking good. 
We can merge devguide PR https://github.com/python/devguide/pull/333 once this has been merged.

**Review by Mariatta — APPROVED** [review]

**Comment by jonparrott** [comment]
Thanks, Mariatta! 

Does anything else need to be done here? Let's merge?

**Comment by ned-deily** [comment]
Thanks Jon and everyone else who worked on this.  Let's see how it works for 3.8.  (I don't think this should be backported.)

**Comment by jonparrott** [comment]
Yay!

**Review by MFredrich — COMMENTED** [review]
Duplicate of #

**Review by MFredrich — COMMENTED** [review]
Duplicate of #

**Pull request merged**
